### PR TITLE
feat(APPT-SQTA-34): View All Vet Schedules and Find schedule by vet Id

### PR DESCRIPTION
**Pull request by GSwizzy**
Added on vetList the functionality to view all Schedules and find a schedule by vet Id.

### CODE REVIEW

File: src/main/resources/db/hsqldb/initDB.sql [hunk 1]
```diff
@@ -1,17 +1,19 @@
 DROP TABLE vet_specialties IF EXISTS;
 DROP TABLE vet_schedule IF EXISTS;
 DROP TABLE vets IF EXISTS;
+DROP TABLE vet_schedule IF EXISTS;
```
Comment (sylveon) [inline src/main/resources/db/hsqldb/initDB.sql]: duplicate line

**Review by sylveon — DISMISSED** [review]
Looks generally OK, you have to fix CI tho

File: src/main/java/org/springframework/samples/petclinic/repository/jdbc/JdbcScheduleRepositoryImpl.java [hunk 2]
```diff
@@ -35,18 +36,32 @@ public JdbcScheduleRepositoryImpl(JdbcTemplate jdbcTemplate) {
...
+    //ERROR IN SQL
     @Override
     public Schedule findScheduleById(int id) {
-        return null;
+        Schedule schedule;
+        try {
+            Map<String, Object> params = new HashMap<>();
+            params.put("vet_id", id);
+            schedule = this.namedParameterJdbcTemplate.queryForObject(
+                "SELECT vet_id, day_available FROM vet_schedule WHERE vet_id=: id",
```
Comment (sylveon) [inline src/main/java/org/springframework/samples/petclinic/repository/jdbc/JdbcScheduleRepositoryImpl.java]:
```suggestion
                "SELECT vet_id, day_available FROM vet_schedule WHERE vet_id = :vet_id",
```

File: src/main/java/org/springframework/samples/petclinic/repository/ScheduleRepository.java [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 public interface ScheduleRepository {
 
-    List<Schedule> listSchedules();
+    List<Schedule> findAllSchedules();
```
Comment (sylveon) [inline src/main/java/org/springframework/samples/petclinic/repository/ScheduleRepository.java]: this should just be `findAll`

File: src/main/java/org/springframework/samples/petclinic/repository/springdatajpa/SpringDataScheduleRepository.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package org.springframework.samples.petclinic.repository.springdatajpa;
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.Repository;
+import org.springframework.samples.petclinic.model.Schedule;
+import org.springframework.samples.petclinic.repository.ScheduleRepository;
+
+import java.util.List;
+
+public interface SpringDataScheduleRepository extends ScheduleRepository, Repository<Schedule, Integer> {
```
Comment (sylveon) [inline src/main/java/org/springframework/samples/petclinic/repository/springdatajpa/SpringDataScheduleRepository.java:10]: this should be a completely empty interface

File: src/main/java/org/springframework/samples/petclinic/repository/jpa/JpaScheduleRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+@Repository
+public class JpaScheduleRepositoryImpl implements ScheduleRepository {
+
+    @PersistenceContext
+    private EntityManager em;
+
+
+    @Override
+    public List<Schedule> findAllSchedules() {
+        return this.em.createQuery("SELECT new vet_schedule (vetId, dayAvailable) FROM vet_schedule").getResultList();
```
Comment (sylveon) [inline src/main/java/org/springframework/samples/petclinic/repository/jpa/JpaScheduleRepositoryImpl.java]:
```suggestion
        return this.em.createQuery("SELECT distinct sched FROM Schedule sched ORDER BY sched.id").getResultList();
```

File: src/main/java/org/springframework/samples/petclinic/repository/jpa/JpaScheduleRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    @Override
+    public Schedule findScheduleById(int id) {
+        Query query = this.em.createQuery("SELECT new vet_schedule (vetId, dayAvailable) FROM vet_schedule WHERE vetId =: id");
```
Comment (sylveon) [inline src/main/java/org/springframework/samples/petclinic/repository/jpa/JpaScheduleRepositoryImpl.java]: Replace this whole thing by `return this.em.find(Schedule.class, id);`

**Review by sylveon — CHANGES_REQUESTED** [review]

File: src/main/java/org/springframework/samples/petclinic/repository/springdatajpa/SpringDataScheduleRepository.java (hunk 4 above)
Comment (sylveon) [reply src/main/java/org/springframework/samples/petclinic/repository/springdatajpa/SpringDataScheduleRepository.java:10]: Nevermind.

**Review by sylveon — APPROVED** [review]

**Review by Colorin09 — APPROVED** [review]

**Pull request merged**